### PR TITLE
Advanced search implementation

### PR DESCRIPTION
**Pull request by rosolowski**
Issue: edmcouncil/onto-viewer#133 

Desktop version of advanced search is ready.

Current behavior:
 - search is now using the "Find search" API [(documentation link)](https://github.com/edmcouncil/onto-viewer/blob/develop/api-doc/rest_api.md#10-find-search).
 - if the "advanced tab" is collapsed - basic mode is used
 - if the "advanced tab" is expanded and at least one property is selected or useHighlihting option toggle is modified - advanced mode is used, otherwise basic mode is still used

Work in progress:
- ~~the properties to select from are not available on the server so example response from documentation is used for presentation purposes until properties are available~~
- mobile version needs to be implemented
- ~~some search terms result in an error, specified in onto-viewer issue [(here)](https://github.com/edmcouncil/onto-viewer/issues/160)~~ <- this is now fixed
- maturity level of search results is not provided in the API

Preview:
image

### CODE REVIEW

**Comment by rosolowski** [comment]
Changes:

- properties are now fetched from the server and RDFS Label is selected by default
- "advanced search" renamed to "search configuration"
- search is now **always** looking at selected properties
- highlights now show their property label

**Review by patrycjamia — CHANGES_REQUESTED** [review]
loupe not working
image

File: fibo/src/styles/setup/_bootstrap-override.scss [hunk 1]
```diff
@@ -1,5 +1,5 @@
 body {
-  font-family: "Roboto", sans-serif;
+  font-family: "Inter", sans-serif;
```
Comment (dasawanaka) [inline fibo/src/styles/setup/_bootstrap-override.scss:2]: Is there any reason to change the font in the project to this one?

Comment (rosolowski) [reply fibo/src/styles/setup/_bootstrap-override.scss:2]: > Is there any reason to change the font in the project to this one?

It's the font used in figma design and it is used in ontology view already but not in other pages, this is update to the global style. Currently you can see the difference in production:
https://spec.edmcouncil.org/fibo/ontology (Inter)
https://spec.edmcouncil.org/fibo/ (Roboto - old font)

**Comment by dasawanaka** [comment]
@mereolog  The auto ontology is also to include a new search mechanism, or is it to stay with the old one?  
If it is to include the new mechanism, there are also required changes to the pages in the auto directory. Remember that auto pages were not refreshed and are on the old gui.

**Comment by mereolog** [comment]
> @mereolog The auto ontology is also to include a new search mechanism, or is it to stay with the old one?
> If it is to include the new mechanism, there are also required changes to the pages in the auto directory. Remember that auto pages were not refreshed and are on the old gui.

Good point - see edmcouncil/html-pages#142.

**Comment by rosolowski** [comment]
> loupe not working 
> image

Added search with loupe.

other changes to search:

- if user enters something to the search box input it will be visible even if search box is no longer active (so that user always knows what is in the search box). When the input is empty the 'Find domains, ontologies, concepts...' placeholder is shown.
- search phrase is now kept in the input field after search instead of keeping it only as placeholder

**Comment by rosolowski** [comment]
Hints are now using find api as mentioned in https://github.com/edmcouncil/onto-viewer/issues/170

**Review by patrycjamia — DISMISSED** [review]

**Review by mereolog — CHANGES_REQUESTED** [review]
When the user did not type any text into the search window, the "List is empty" box/window of results is displayed:
image
When displayed it hides the heading 'search configuration' and sort of distorts the display.

If this is easy, could we change it so that when the user does not type any text, no "result list" window will be displayed?

**Comment by rosolowski** [comment]
> When the user did not type any text into the search window, the "List is empty" box/window of results is displayed: image When displayed it hides the heading 'search configuration' and sort of distorts the display.
> 
> If this is easy, could we change it so that when the user does not type any text, no "result list" window will be displayed?

Changes:
- Search dropdown is now hidden when input field is empty.

**Review by mereolog — APPROVED** [review]

**Pull request merged**